File: Sarvam C-DAC Project/Sarvam C-DAC Spring Boot/Sarvam/src/main/java/com/cdac/sarvam/model/MasterLeaveRequest.java
package com.cdac.sarvam.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import jakarta.persistence.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "master_leave_request")
public class MasterLeaveRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "request_id")
    private Long requestId;

    @ManyToOne
    @JoinColumn(name = "prn", referencedColumnName = "PRN", nullable = false)
    private MasterProfile masterProfile; // Foreign key reference to MasterProfile

    @Column(name = "from_date", nullable = false)
    private LocalDate fromDate;

    @Column(name = "end_date", nullable = false)
    private LocalDate endDate;

    @Column(name = "subject", length = 255, nullable = false)
    private String subject;

    @Column(name = "description", length = 255)
    private String description;

    @Column(name = "status", nullable = false)
    private String status = "Pending"; // Default value

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();
}

File: MS.NET/Sarvam Backend/Models/MasterLeaveRequest.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

[Table("master_leave_request")]
public class MasterLeaveRequest
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("request_id")]
    public int RequestId { get; set; }

    [ForeignKey("MasterProfile")]
    [Column("prn")]
    public long Prn { get; set; }

    [Required]
    [Column("from_date")]
    public DateTime FromDate { get; set; }

    [Required]
    [Column("end_date")]
    public DateTime EndDate { get; set; }

    [Required]
    [StringLength(255)]
    [Column("subject")]
    public string Subject { get; set; }

    [StringLength(255)]
    [Column("description")]
    public string Description { get; set; }

    [Required]
    [Column("status")]
    public string Status { get; set; } = "Pending";

    [Required]
    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    [Required]
    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.Now;

    // Ensuring that the end_date is greater than from_date can be enforced in EF Core using Fluent API
    // which is explained below in OnModelCreating method of the DbContext class.
}
